evaluation_experiments/EXP3.py

```diff
@@ -0,0 +1,238 @@
+   files_path = files_path, noise_type = noise_type, cibersort_files = cibersort_files, 
+   random_seed = rs, num_samples = num_samples, nmf_cut = nmf_cut_value, kernel_name = ker)
+)
+####################################################################################################
+
+#Excecuting notebook with noise.
+#random
+noise_type = 'noise'
+bulkprop_type = "random"
+pm.execute_notebook(
+   in_nb,
+   f'{results_path}/EXP3_{data}_{noise_type}_{bulkprop_type}.ipynb', kernel_name= ker,
+   parameters=dict( res_name = res_name, pseudos_name = pseudos_name, 
+   files_path = files_path, noise_type = noise_type, cibersort_files = cibersort_files, 
+   random_seed = rs, num_samples = num_samples, nmf_cut = nmf_cut_value, kernel_name = ker)
+)
+#realistic
+noise_type = 'noise'
+bulkprop_type = "realistic"
+pm.execute_notebook(
+   in_nb,
+   f'{results_path}/EXP3_{data}_{noise_type}_{bulkprop_type}.ipynb', kernel_name= ker,
+   parameters=dict( res_name = res_name, pseudos_name = pseudos_name, 
+   files_path = files_path, noise_type = noise_type, cibersort_files = cibersort_files, 
+   random_seed = rs, num_samples = num_samples, nmf_cut = nmf_cut_value, kernel_name = ker)
+)
+
+####################################################################################################
+####################################################################################################
+
+#BayesPrism now
+
+#notebook to run
+in_nb = f'{a_path}/evaluation_experiments/EXP3_BayesPrism_eval.ipynb'
+
+#####################
+### set the study ###
+#####################
+#paths:
+data = "adp" #adipose data single cell and single nucleus
+res_name = f"MCT_adp_EXP3"
+pseudos_name = "MCT_adp_EXP3"
+files_path = "/data/EXP3/"
+###### set your random seed, num missing cells, and num_samples for reference for reproducibility
+rs = 88 #random seed
+num_samples = 10000
+#how to handle negative residual distribution for NMF
+nmf_cut_value = "minimum_value"
+kerne_name = ker
+
+###################################################################################################
+
+#Excecuting notebook with no noise.
+noise_type = 'nonoise'
+bulkprop_type = "random"
+pm.execute_notebook(
+   in_nb,
+   f'{results_path}/EXP3_{data}_{bulkprop_type}_{noise_type}.ipynb', kernel_name= ker,
+   parameters=dict( res_name = res_name, pseudos_name = pseudos_name, 
+   files_path = files_path, noise_type = noise_type, cibersort_files = cibersort_files, 
+   random_seed = rs, num_samples = num_samples, nmf_cut = nmf_cut_value, kernel_name = ker)
+)
+
+noise_type = 'nonoise'
+bulkprop_type = "realistic"
+pm.execute_notebook(
+   in_nb,
+   f'{results_path}/EXP3_{data}_{bulkprop_type}_{noise_type}.ipynb', kernel_name= ker,
+   parameters=dict( res_name = res_name, pseudos_name = pseudos_name, 
+   files_path = files_path, noise_type = noise_type, cibersort_files = cibersort_files, 
+   random_seed = rs, num_samples = num_samples, nmf_cut = nmf_cut_value, kernel_name = ker)
+)
+
```

Comment:
do you run this entire script on Alpine? You could run each of these papermill jobs as an independent job to parallelize the computation.

Reply:
this is a good idea but i dont expect people to run them all at a time, they're kind of sequential since bayesprism and cibersort results are saved and reloaded in the _eval. notebook. the readme says this.